Training time window 1
Waiting for JIT...
Device: gpu
Traceback (most recent call last):
  File "/home/david/work/repos/deep-learning-pde/jaxpi/examples/cahn_hilliard/main.py", line 42, in <module>
    app.run(main)
    ~~~~~~~^^^^^^
  File "/home/david/anaconda3/envs/jaxpi/lib/python3.13/site-packages/absl/app.py", line 308, in run
    _run_main(main, args)
    ~~~~~~~~~^^^^^^^^^^^^
  File "/home/david/anaconda3/envs/jaxpi/lib/python3.13/site-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
             ~~~~^^^^^^
  File "/home/david/work/repos/deep-learning-pde/jaxpi/examples/cahn_hilliard/main.py", line 34, in main
    train.train_and_evaluate(FLAGS.config, FLAGS.workdir)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/david/work/repos/deep-learning-pde/jaxpi/examples/cahn_hilliard/train.py", line 109, in train_and_evaluate
    model = train_one_window(config, workdir, model, samplers, u, idx)
  File "/home/david/work/repos/deep-learning-pde/jaxpi/examples/cahn_hilliard/train.py", line 38, in train_one_window
    model.state = model.step(model.state, batch)
                  ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/home/david/anaconda3/envs/jaxpi/lib/python3.13/site-packages/jaxpi/models.py", line 171, in step
    grads = grad(self.loss)(state.params, state.weights, batch, *args)
  File "/home/david/anaconda3/envs/jaxpi/lib/python3.13/site-packages/jaxpi/models.py", line 126, in loss
    weighted_losses = tree_map(lambda x, y: x * y, losses, weights)
ValueError: Dict key mismatch; expected keys: ['bc', 'ics', 'res']; dict: {'ics': Traced<ShapedArray(float32[], weak_type=True)>with<DynamicJaxprTrace>, 'res': Traced<ShapedArray(float32[], weak_type=True)>with<DynamicJaxprTrace>}.
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
